### ISSUE TITLE
VS code block indent lines off by one character

### ISSUE THREAD

**Issue by vaindil** [aspnet/Razor#2035]
Not sure if this should be opened here or on the VS feedback tracker, let me know.

I'm on the latest version of everything, so VS 15.5.6, ASP.NET Core 2.0, .NET Core SDK 2.1.4. This has been a problem for months now, I don't remember exactly when it started and I couldn't find another issue here when I searched. Most indent lines in VS Razor views are off by one character, but not all of them.

`if` is off by one, but the `else` is correct.
![image](https://user-images.githubusercontent.com/771134/35974790-741e1e7e-0ca8-11e8-9c4f-2a37630f6b9f.png)

Both `foreach` and `for` are off.
![image](https://user-images.githubusercontent.com/771134/35975161-bce116b0-0ca9-11e8-9929-6b486af3f147.png)

`do`/`while` is off.
![image](https://user-images.githubusercontent.com/771134/35975250-0e5d819a-0caa-11e8-8bce-adb6947b4925.png)

**Comment by mkArtakMSFT** [comment]
Thanks for contacting us, @vaindil.
@NTaylorMullen, any thoughts regarding this?

**Comment by NTaylorMullen** [comment]
@vaindil thanks for the report! We previously had an issue created by someone on the team about this [here](https://github.com/aspnet/Razor/issues/1257) but after investigation determined the cost was too high to fix. @mkArtakMSFT given that there is more reports of this issue we could either close as a wont fix (since that's what we did before) or put it in the backlog.

For referene here was the result of the previous investigation:

> Was able to repro. So we could potentially fix this but the cost would be a misaligned debugging and possibly formatting experience. Let me try and explain:
> 
> The if part of the statement dictates where the C# editor places the vertical guide. Given that there's an @ in front of the if it moves the vertical guidelines 1 character forward. We could offset the indentation of the underlying braces of an @if statement by 1 character to account for this; however, that would result in the yellow debugging highlights appearing 1 character off from where they should be. It might also impact formatting, we've had similar things impact it in the past. We typically render code behind the scenes at the exact character location as seen in the editor to make debugging and formatting behave as expected.
> 
> tl;dr; I don't think it's worth doing unless other teams get involved to fix all the issues (debugging, formatting etc).

**Comment by vaindil** [comment]
@NTaylorMullen Gotcha, that makes sense. It previously worked properly though. I don't remember when the problem started, but I'd been on ASP.NET Core with it working properly and then the issue started. Oh well, no big deal overall. Thank you!

**Comment by NTaylorMullen** [comment]
> It previously worked properly though. I don't remember when the problem started, but I'd been on ASP.NET Core with it working properly and then the issue started.

That's really valuable information actually. Probably means that vertical guide code (we don't own) used to behave differently and was changed.

**Comment by vaindil** [comment]
I'll have some free time in the next few days, if there's a way to download old versions of VS 2017 I have no problem spending a while figuring out exactly when the issue started. I have some TV shows to catch up on anyway. :P

**Comment by NTaylorMullen** [comment]
> if there's a way to download old versions of VS 2017

I'm personally not aware of an official way to do this haha. It's quite awesome of you to offer your time however. If this gets more traction then we can revisit.

**Comment by mkArtakMSFT** [comment]
@NTaylorMullen, is this something we have control of, or is this something the tooling team should handle?
